primitive-types/impls/serde/src/serialize.rs

```diff
@@ -32,7 +33,8 @@ fn to_hex<'a>(v: &'a mut [u8], bytes: &[u8], skip_leading_zero: bool) -> &'a str
 		idx += 2;
 	}
 
-	::std::str::from_utf8(&v[0..idx]).expect("All characters are coming from CHARS")
+	// SAFETY: all characters come from CHARS array.
```

Comment:
nitpick: this isn't true, the `v` is prepended with `0x`

Reply:
I suggest something `all characters are ASCII, therefore valid UTF8`

Reply:
```suggestion
	// SAFETY: all characters come either from CHARS or "0x", therefore valid UTF8
```